server/tests/test_soft_delete.py

```diff
@@ -0,0 +1,194 @@
+    record0_25 = indicator.data.create(
+        period=period,
+        dimension_type=age_dim_type,
+        literal_dimension_val="0-25",
+        value=5,
+    )
+    record25_50 = indicator.data.create(
+        period=period,
+        dimension_type=age_dim_type,
+        literal_dimension_val="25-50",
+        value=6,
+    )
+    record_51_75 = indicator.data.create(
+        period=period,
+        dimension_type=age_dim_type,
+        literal_dimension_val="51-75",
+        value=7,
+    )
+
+    # soft-delete the first datum
+    data = {
+        "predefined-TOTAL_FORMS": 0,
+        "predefined-INITIAL_FORMS": 0,
+        "predefined-MIN_NUM_FORMS": 0,
+        "predefined-MAX_NUM_FORMS": 1000,
+        "agegroup-TOTAL_FORMS": 3,
+        "agegroup-INITIAL_FORMS": 3,
+        "agegroup-MIN_NUM_FORMS": 0,
+        "agegroup-MAX_NUM_FORMS": 1000,
+        "agegroup-0-id": record0_25.id,
+        "agegroup-0-literal_dimension_val": "0-25",
+        "agegroup-0-value": 5.5,
+        "agegroup-1-id": record25_50.id,
+        "agegroup-1-literal_dimension_val": "25-50",
+        "agegroup-1-value": 7.5,
+        # delete record_51_75
+        "agegroup-2-id": record_51_75.id,
+        "agegroup-2-literal_dimension_val": "51-75",
+        "agegroup-2-is_deleted": "on",
+    }
+    url = reverse(
+        "manage_indicator_data",
+        args=[indicator.id, period.id, age_dim_type.pk],
+    )
+
+    with patch_rules(can_edit_indicator_data=True):
+        response = vanilla_user_client.post(url, data=data)
+        assert response.status_code == 302
+
+    # test object manager should show all objects; including deleted ones
+    assert (
+        IndicatorDatum.objects.filter(
+            indicator=indicator, period=period
+        ).count()
+        == 3
+    )
+    # test active objects should only show not deleted items
+    assert (
+        IndicatorDatum.active_objects.filter(
+            indicator=indicator, period=period
+        ).count()
+        == 2
+    )
+
+    # submit all data as program including deleted data
+    url = reverse("submit_indicator_data_all", args=[indicator_id, period.id])
+    with patch_rules(can_submit_as_hso_or_program=True):
+        resp = vanilla_user_client.post(url, {"submission_type": "program"})
+        assert resp.status_code == 302
+
+    statuses = get_submission_statuses(indicator, period)
+
+    # submit all data as hso including deleted data
+    assert statuses == {
+        "statuses_by_dimension_type_id": {
+            age_dim_type.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+        },
+        "global_status": SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+    }
+
+    url = reverse("submit_indicator_data_all", args=[indicator_id, period.id])
+    with patch_rules(can_submit_as_hso_or_program=True):
+        resp = vanilla_user_client.post(url, {"submission_type": "hso"})
+        assert resp.status_code == 302
+
+    statuses = get_submission_statuses(indicator, period)
+
```

Comment:
This could be a good place to assert the version has `is_deleted=True, program_submitted=True, hso_submitted=True`